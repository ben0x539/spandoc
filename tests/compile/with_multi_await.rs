use spandoc::spandoc;

#[spandoc]
fn main() {
    let fut = async {
        4usize
    };

    let fut2 = async {
        /// do the thing
        fut.await
    };

    let fut3 = async {
        5usize
    };

    /// this should fail
    let fut4 = async {
        let _ = fut2.await;
        fut3.await
    };
}
