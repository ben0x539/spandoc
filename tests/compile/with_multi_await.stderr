error: spandoc does not support instrumenting multiple awaits with a single span
  --> $DIR/with_multi_await.rs:21:9
   |
21 |         fut3.await
   |         ^^^^^^^^^^
